app/models/miq_storage_metric.rb

```diff
@@ -183,20 +180,13 @@ def self.purge_daily_metrics_rollups_by_date(older_than, window = nil)
 
   def self.purge(older_than, rollup_type, window, metrics_classes)
     window ||= purge_window_size
-    gtotal = 0
-    metrics_classes.each do |mc|
-      conditions = mc.arel_table[:statistic_time].lt(older_than)
-      query = mc.select(:id).where(conditions)
+    metrics_classes.map do |mc|
+      query = mc.where(mc.arel_table[:statistic_time].lt(older_than))
       query = query.where(:rollup_type => rollup_type) unless rollup_type.nil?
-      total = 0
-      query.find_in_batches(:batch_size => window) do |ma|
-        ids = ma.collect(&:id)
-        total += mc.delete_all(:id => ids)
+      delete_in_batches(query, window).tap do |total|
```

Comment:
I might be wrong here, but shouldn't it be `delete_in_batches(window)`? or `query.delete_in_batches(window)`?